lib/improver/utilities/rescale.py

```diff
@@ -81,3 +81,40 @@ def rescale(data, data_range=None, scale_range=(0., 1.),
     if clip:
         result = np.clip(result, scale_min, scale_max)
     return result
+
+
+def apply_double_scaling(data_cube, scaled_cube,
+                         data_vals, scaling_vals,
+                         combine_function=np.minimum):
+    """
+    Update scaled_cube based on the contents of data_cube so that
+    scaled_cube is at least the value of the data_cube after rescaling
+    based on an upper, mid and lower threshold.
```

Comment:
This isn't what this function does.  I'm not 100% sure but what about something like:
```
Define a grid of limiting values based on data values at a point.  Then
scale data in another cube according to these limits using the
combine_function.  Default sets scaled_cube.data to be below the
local_limit calculated from data_cube.
```